modules/auxiliary/gather/query_ldap.rb

```diff
@@ -0,0 +1,146 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+
+  include Msf::Exploit::Remote::LDAP
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'LDAP Query and Enumeration Module',
+        'Description' => %q{
+          This module allows users to query an LDAP server using either a custom LDAP query, or a set of LDAP queries under a specific category.
+          The custom query is controlled via the LDAPQUERY parameter, or if one wants to run a set of predefined queries they can use the PREDEFINEDQUERY
+          option to specify a set of predefined queries to run.
+
+          All results will be returned to the user as plain text.
+        },
+        'Author' => [
+          'Grant Willcox', # Module
+        ],
+        'References' => [
+        ],
+        'DisclosureDate' => '2022-05-19',
+        'License' => MSF_LICENSE,
+        'DefaultOptions' => {
+          'SSL' => false
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'SideEffects' => [IOC_IN_LOGS],
+          'Reliability' => []
+        }
+      )
+    )
+
+    register_options([
+      Opt::RPORT(389), # Set to 636 for SSL/TLS
+      OptString.new('BASE_DN', [false, 'LDAP base DN if you already have it']),
+      OptString.new('LDAPQUERY', [false, 'Query to run against the target LDAP server']),
+      OptEnum.new('PREDEFINEDQUERY', [false, 'Predefined query to run', nil, ['ALL', 'USERS', 'PEOPLE', 'GROUPS', 'ORGUNITS', 'ORGROLES', 'COMPUTERS', 'EXCHANGE'] ])
+    ])
+  end
+
+  def perform_ldap_query(ldap, filter, entries)
+    returned_entries = ldap.search(base: @base_dn, filter: filter)
+    p returned_entries
```

Comment:
Is this meant to be printing directly to stdout? Normally we use the `print_*` methods to ensure that framework handles logging.

Reply:
This is for debugging and is in no way intended to be final. I'm interested in soliciting feedback on what the best way to print this out would be.

Reply:
As per Spencer's feedback decided to go with Rex::Table output for the time being. Implemented in https://github.com/rapid7/metasploit-framework/pull/16598/commits/b6c7aea988ffc1f77b83a4d5caa892c69c696ad6